### DISCUSSION TITLE
New formula / algorithm for the Eltako wind sensor ("anemometer") in newer tasmota versions

### DISCUSSION THREAD

**Discussion by jayjupdhig** [arendst/Tasmota#12453] [Ideas]
Hi there

That one does not work anymore - compared to older (e.g. 5.8 or 6.5) versions:

```
double value = Double.parseDouble(clWindValue);

			if (value == 0.01d) {
				value = lastValue;
			} else {
				if (value == 64.999d) {
					value = 0d;
				} else {
					if (value > 0.01d) {
						value = Math.round(1.761d / (1 + (1 / (value * 2))) + 3.013d * (1 / (value * 2)) * 100.0) / 100.0;
					}
				}
			}

			if (lastValue == -1d || lastValue != value) {
				lastValue = value;
				System.out.println("Windgeschwindigkeit: " + lastValue);
			}
```

-> ´clWindValue´ is the value i get from JSON...
URL: http://192.168.254.149/cm?cmnd=counter

Configured as:

![ttt](https://user-images.githubusercontent.com/42765391/123084089-09e5da80-d421-11eb-9385-4e255b9baf8a.JPG)

AND, additionally (AND necessary!!) executed in Tasmota console: ´CounterType1 1´

IF you need more informations, so please don't heisitate to ask me.

Thank you very much for your feedbacks.

**Comment by sfromis** [comment]
Well, that Java code there is not useful for guessing what may be different. What `clWindValue` values do you get now, and what did you get on the older versions? Current format is JSON: `{"Counter1":12345}`

**Reply by jayjupdhig** [reply]
Thank you.

Whole Code here, but it's for the previous method (with older Tasmota Version, e.g. 5.8. or 6.5)

The problem is that i have NO IDEA what has changed in Tasmota internals, sampling frequency or somthing like that. (???)

However, with older version i get the correct wind speed value with that formula, for that case it was correct. But since they released newer versions (>=8.x oder >=7.x...? Currently, we have 9.1) internally something has changed...

Whole code (compatible with older versions but not with newer ones) here:

```
private void start(final URL obj) throws Exception {
		double lastValue = -1;
		while (true) {
			HttpURLConnection con = (HttpURLConnection) obj.openConnection();
			BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
			String inputLine;

			String response;

			StringBuffer responseStringBuffer = new StringBuffer();

			while ((inputLine = in.readLine()) != null) {
				responseStringBuffer.append(inputLine);
			}

			response = responseStringBuffer.toString();

			in.close();

			final JSONParser clJSONParser = new JSONParser();

			final JSONObject clJSONObject = (JSONObject) clJSONParser.parse(response);

			final String clWindValue = clJSONObject.get("Counter1").toString();

			// System.out.println("Counter1: " + clWindValue);

			/*
			final Matcher matcher = pattern.matcher(response);
			matcher.find();
			
			final String match = matcher.group(1);
			
			double value = Double.parseDouble(match.substring(0, match.length() - 4));
			*/

			double value = Double.parseDouble(clWindValue);

			if (value == 0.01d) {
				value = lastValue;
			} else {
				if (value == 64.999d) {
					value = 0d;
				} else {
					if (value > 0.01d) {
						value = Math.round(1.761d / (1 + (1 / (value * 2))) + 3.013d * (1 / (value * 2)) * 100.0) / 100.0;
					}
				}
			}

			if (lastValue == -1d || lastValue != value) {
				lastValue = value;
				System.out.println("Windgeschwindigkeit: " + lastValue);
			}

			Thread.sleep(1000);
		}
	}
```

**Reply by jayjupdhig** [reply]
Another question: What do i have to use for code blocks in GitHub...?

That does also not seem to work anymore:

![ccc](https://user-images.githubusercontent.com/42765391/123094782-99918600-d42d-11eb-8dc4-666008e5d95d.JPG)

**Reply by jayjupdhig** [reply]
Oh - it works!! ;-)

The "problem" was that the background color is not white anymore... but there are some scroll bars at least!! ;-)

**Reply by barbudor** [reply]
You need to tell us which is the exact last version working and which is the 1st version in which it is broken.

**Reply by sfromis** [reply]
You did not answer my question about what data you got before and now. Those simple bits of info is more useful than looking a your Java code.

**Reply by jayjupdhig** [reply]
Thank you very much for your feedbacks.

Problem seems to be solved - it only one thing that has changed seems to be the floating point values...

(But i'm still not sure about the ´CounterDebounce´ value for the Eltako wind sensor; seems to be 300 in general...(?))

**Comment by jayjupdhig** [comment]
OK thx, problem solved... :-)
